piker/brokers/kraken.py

```diff
@@ -308,34 +300,38 @@ async def get_trades(
 
     async def submit_limit(
         self,
-        oid: str,
         symbol: str,
         price: float,
         action: str,
         size: float,
-        reqid: int = None,
-    ) -> int:
+        reqid: str = None,
+        validate: bool = False # set True test call without a real submission
+    ) -> dict:
         '''
         Place an order and return integer request id provided by client.
 
         '''
-        # Build order data for kraken api
+        # Build common data dict for common keys from both endpoints
         data = {
-            "userref": reqid,
-            "ordertype": "limit",
-            "type": action,
-            "volume": str(size),
             "pair": symbol,
             "price": str(price),
-            # set to True test AddOrder call without a real submission
-            "validate": False
+            "validate": validate
         }
-        return await self.endpoint('AddOrder', data)
+        if reqid is None:
+            # Build order data for kraken api
+            data["ordertype"] = "limit"
```

Comment:
fwiw this is a good spot to use the new merge syntax:

`data |= {'ordertype': 'limit', 'type': action, 'volume': str(size)}`

if you wanted but nbd and it's def not a requirement for this pr.

Reply:
oh and i also prefer the name `payload` for inputs to endpoints just for the next time.

Reply:
oh slick, I like that new syntax, I will update it. I can also switch data to payload if that is consistent with the rest of the base

Reply:
acutally given ill leave the payload change for next time. <3